matsim/src/main/java/org/matsim/core/events/ParallelEventsManager.java

```diff
@@ -82,7 +82,8 @@ public ParallelEventsManager(final boolean syncOnTimeSteps, final int numOfThrea
 		this.eventsHandlers = new ArrayList<EventHandler>();
 		this.eventsArraySize = syncOnTimeSteps ? 512 : 32768;
 		this.eventsQueueSize = eventsQueueSize;
-		this.eventQueue = new ArrayBlockingQueue<>(eventsQueueSize);
+//		this.eventQueue = new ArrayBlockingQueue<>(eventsQueueSize);
+		this.eventQueue = new ArrayBlockingQueue<>(1048576 * 64);
```

Comment:
provide the queue size via a constructor - this should solve the problem without hardcoding the queue size.